src/components/Custom/RJSFFormFieldAdapter/CustomPriorityBoundsField.jsx

```diff
@@ -169,12 +172,140 @@ const CustomPriorityBoundsField = (props) => {
     }
   }, [isMapVisible]);
 
+  // Close tooltip when clicking outside
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (showTooltip && !event.target.closest(".tooltip-container")) {
+        setShowTooltip(false);
+      }
+    };
+
+    document.addEventListener("click", handleClickOutside);
+    return () => document.removeEventListener("click", handleClickOutside);
+  }, [showTooltip]);
+
   const handleChange = (newData) => {
     if (typeof props.onChange === "function") {
       props.onChange(Array.isArray(newData) ? [...newData] : newData);
     }
   };
 
+  // Validate GeoJSON structure
+  const validateGeoJSON = (geoJson) => {
+    if (!geoJson || typeof geoJson !== "object") {
+      return { valid: false, error: "Invalid JSON format" };
+    }
+
+    if (!Array.isArray(geoJson.features)) {
+      return { valid: false, error: "Features must be an array" };
+    }
+
+    const polygonFeatures = geoJson.features.filter(
+      (feature) =>
+        feature.type === "Feature" && feature.geometry && feature.geometry.type === "Polygon",
+    );
+
+    if (polygonFeatures.length === 0) {
+      return { valid: false, error: "No valid Polygon features found" };
+    }
+
+    return { valid: true, features: polygonFeatures };
+  };
```

Comment:
This function is kind of wrong; it doesn't actually require the input to be valid GeoJSON. I think the correct thing to do is to expect a GeoJSON `Feature` or `FeatureCollection`. Turf has helpers for validating GeoJSON; we should use those.